### ISSUE TITLE
Allow only specific users (no group)

### ISSUE THREAD

**Issue by dwasyl** [django-auth-ldap/django-auth-ldap#57]
I have a use case for allowing only a selected set of users. There isn't the flexibility to create a group in the AD, ideally users would be 'selected' by an admin user. Either from a directory listing, or manually typed in. Is it possible to set up something like this using this package? Basically, creating a sort of group on the Django side.

Any ideas?

**Comment by jdufresne** [comment]
You can extend `LDAPBackend` and override `authenticate()` to do additional checks on `user` after the LDAP authentication. Would this solve your use case?

**Comment by dwasyl** [comment]
Perhaps, that could be used to check for an existing Django account (and only allow authentication if it exists or if there's a matching property).

As for accessing a directory of users, to select who is allowed, is that possible with the `auth-ldap`, or does that need to be through some other method?

**Comment by jdufresne** [comment]
Does this section on using groups to limit access help? Otherwise, I'm a bit confused as to what you're looking for.

https://django-auth-ldap.readthedocs.io/en/latest/groups.html#limiting-access

If you're still having issues, perhaps begin using the library to discover what it is capable of. If you can offer a PR that improves your use case I'll happily evaluate it. Or perhaps point to the missing feature.

**Comment by kitor** [comment]
> Does this section on using groups to limit access help?
I seem to have the same problem. In corporate env I have access to LDAP as auth service, but we cannot use / assign / create groups via LDAP, we are assigning rights per user login in each app (eg. in Jenkins). 

On auth it ends with
```
search_s('YYY') returned 1 objects: XXX
Populating Django user XXX
Caught LDAPError while authenticating XXX: INSUFFICIENT_ACCESS({'desc': 'Insufficient access'},)
```

**Comment by vilhelmen** [comment]
Hi there, I think I actually have the same issue. I don't have the authority to create a group to use with AUTH_LDAP_REQUIRE_GROUP for one of our sites (ugh!), and I'm looking to whitelist domain users via the Django user database.

Looking at [backend.py line 609](https://github.com/django-auth-ldap/django-auth-ldap/blob/master/django_auth_ldap/backend.py#L609), all you'd need to do is deny and built user, so something like the following on line 112 or 114:

```
if built and self.settings.EXISTING_USERS_ONLY:
    raise UserNotAllowedException(self._username + " is not in local users database!")
```

I can flesh this out with some docs and tests and issue a pull if you'd like! I'll probably implement it anyway for our use.

**Comment by jdufresne** [comment]
Fixed in eb930761996b216d10156fba4ddf658b3213643b Thanks @vilhelmen !
